.github/workflows/backend-test.yml

```diff
@@ -53,6 +53,7 @@ jobs:
           testcoverage="${{ env.coverage }}"
           git fetch origin "$base_branch"
           git checkout "origin/$base_branch"
+          go mod tidy
```

Comment:
Can you please explain this one?

I don't think we should do this here. Because the module system should be working for anyone checking it out. Having this here means we don't test that `go mod tidy` was ran.

Reply:
Make sense. When updating Golang to 1.22, what happens is that https://github.com/headlamp-k8s/headlamp/blob/main/.github/workflows/backend-test.yml#L56 is unable to get the base branch coverage because it is still running on 1.20. I discussed this with Santhosh and the logical way is to update Golang version first in a separate PR and the other things in this PR.